### PR TITLE
Parameterize tests with OpenSearchIntegTestCase.SuiteScopeTestCase annotation

### PR DESCRIPTION
**Pull request by neetikasinghal**
### Description
Parameterize tests annotated with @OpenSearchIntegTestCase.SuiteScopeTestCase.
Without this, the parameterization is not working as initializeSuiteScope() expects a default constructor and we have a different constructor with parameterization.
ref: https://build.ci.opensearch.org/job/gradle-check/24979/testReport/junit/org.opensearch.search.aggregations/MissingValueIT/classMethod/
this has many classes failing with this exception: https://build.ci.opensearch.org/job/gradle-check/24979/#showFailuresLink

### Related Issues
Relates #9443 

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [ ] Commits are signed per the DCO using --signoff
- [ ] Commit changes are listed out in CHANGELOG.md file (See: [Changelog](../blob/main/CONTRIBUTING.md#changelog))

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by neetikasinghal** [comment]
@sohami @reta please review.

**Comment by neetikasinghal** [comment]
> ### Gradle Check (Jenkins) Run Completed with:
> * **RESULT:** FAILURE ❌
> * **URL:** https://build.ci.opensearch.org/job/gradle-check/25081/
> * **CommitID:** [b62a9fb](https://github.com/opensearch-project/OpenSearch/commit/b62a9fbfe9d62c8a43f702aaf50eb3efc42da51f)
>   Please examine the workflow log, locate, and copy-paste the failure(s) below, then iterate to green.
>   Is the failure [a flaky test](https://github.com/opensearch-project/OpenSearch/blob/main/DEVELOPER_GUIDE.md#flaky-tests) unrelated to your change?

https://github.com/opensearch-project/OpenSearch/issues/9815 - this is currently being fixed.

**Comment by sohami** [comment]
I don't have any other suggestion for this change. So LGTM. @reta / @andrross Do you have any feedback for this specific handling ?

**Review by sohami — APPROVED** [review]

File: test/framework/src/main/java/org/opensearch/test/OpenSearchIntegTestCase.java [hunk 1]
```diff
@@ -2317,7 +2317,11 @@ private static void initializeSuiteScope() throws Exception {
         assert INSTANCE == null;
         if (isSuiteScopedTest(targetClass)) {
             // note we need to do this this way to make sure this is reproducible
-            INSTANCE = (OpenSearchIntegTestCase) targetClass.getConstructor().newInstance();
+            if (targetClass.getAnnotation(SuiteScopeTestCase.class).isParameterized()) {
```
Comment (reta) [inline test/framework/src/main/java/org/opensearch/test/OpenSearchIntegTestCase.java:2320]: 👍

Comment (andrross) [reply test/framework/src/main/java/org/opensearch/test/OpenSearchIntegTestCase.java:2320]: Could we just use reflection to check if the test class has a constructor that takes a single `Settings` parameter and if so supply empty settings to it?

Comment (neetikasinghal) [reply test/framework/src/main/java/org/opensearch/test/OpenSearchIntegTestCase.java]: @reta @andrross thanks for the suggestions, i have updated the code to check this by reflection.

**Review by reta — APPROVED** [review]

File: test/framework/src/main/java/org/opensearch/test/OpenSearchIntegTestCase.java [hunk 2]
```diff
@@ -2412,6 +2416,16 @@ private static boolean isSuiteScopedTest(Class<?> clazz) {
         return clazz.getAnnotation(SuiteScopeTestCase.class) != null;
     }
 
+    /*
+    * For tests defined with, SuiteScopeTestCase return true if the
+    * class has a constructor that takes a single Settings parameter
+    * */
+    private static boolean isSuiteScopedTestParameterized(Class<?> clazz) {
```
Comment (andrross) [inline test/framework/src/main/java/org/opensearch/test/OpenSearchIntegTestCase.java:2423]: Nitpick, but there's technically nothing to do with "suite scope" in this method. Something like "hasSettingsConstructor()" or "hasSingleAritySettingsConstructor()" or similar might be better.

**Pull request merged**